### PR TITLE
Added support for JWT based authentication

### PR DESCRIPTION
**Pull request by sukalpomitra**
This PR is to add support for JWT based authentication. The jwt will be extracted from header, validated and a teedy user will be created and injected as principal. For every subsequent call the teedy user creation will be bypassed as the user already exists

### CODE REVIEW

File: docs-web-common/pom.xml [hunk 1]
```diff
@@ -68,6 +68,27 @@
       <groupId>org.slf4j</groupId>
       <artifactId>jul-to-slf4j</artifactId>
     </dependency>
+
+    <dependency>
```
Comment (jendib) [inline docs-web-common/pom.xml:72]: Would it be possible to create a simple http client without openfeign and all the associated dependencies?

File: docs-web-common/src/main/java/com/sismics/util/filter/JwtBasedSecurityFilter.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        try {
+            buildJWTVerifier(jwt).verify(token);
+            // if token is valid no exception will be thrown
+            log.info("Valid TOKEN");
+            return Boolean.TRUE;
+        } catch (CertificateException e) {
+            //if CertificateException comes from buildJWTVerifier()
+            log.info("InValid TOKEN");
+            e.printStackTrace();
```
Comment (jendib) [inline docs-web-common/src/main/java/com/sismics/util/filter/JwtBasedSecurityFilter.java:86]: avoid printStackTrace() and add the exception as log.info() argument

**Review by jendib — CHANGES_REQUESTED** [review]

File: docs-web-common/src/main/java/com/sismics/util/filter/JwtBasedSecurityFilter.java (hunk 2 above)
Comment (sukalpomitra) [reply docs-web-common/src/main/java/com/sismics/util/filter/JwtBasedSecurityFilter.java]: @jendib have updated all printstacktrace with log.info

File: docs-web-common/pom.xml (hunk 1 above)
Comment (sukalpomitra) [reply docs-web-common/pom.xml:72]: @jendib are you suggesting to use resttemplates?

Comment (jendib) [reply docs-web-common/pom.xml:72]: @sukalpomitra OkHttp is already imported and used in `WebhookAsyncListener`. Please use that.

Comment (sukalpomitra) [reply docs-web-common/pom.xml:72]: hi @jendib I have implemented okHttp. Please have a look.

Comment (sukalpomitra) [reply docs-web-common/pom.xml:72]: hi @jendib please check the PR once you have time

**Comment by jendib** [comment]
@sukalpomitra Could you disable this filter by default and add a system property `docs.jwt_authentication` to enable it just like `HeaderBasedSecurityFilter`? It would be safer for all users who don't want to use this auth method.

**Comment by jendib** [comment]
Great thanks 👍

**Pull request merged**